Library/Formula/ice.rb

```diff
@@ -32,7 +32,10 @@ def install
         --enable-cxx
       ]
 
-      args << "--enable-java" if build.with? "java"
+      if build.with? "java"
+        args << "--enable-java"
+        inreplace "dist/Makefile.in", "@JAVACFLAGS@",  "@JAVACFLAGS@ -source 1.7 -target 1.7"
```

Comment:
It does not depend on Java 1.7 though, Java 1.8 builds fine. The issue is if I build with Java 8 (the default if both are installed) then want to use it with Java 7 only project. Which is what this replace fixes.